app/assets/javascripts/helpers/array_of_weekends.js

```diff
@@ -0,0 +1,23 @@
+// Returns array of date strings formatted as 'YYYY-MM-DD'.
+// Each date is a weekend.
+App.Helpers.arrayOfWeekends = function(start, duration) {
+  var count;
+  var date = '',
+      dateRegExMatcher = /^(19|20)\d\d[- /.](0[1-9]|1[012])[- /.](0[1-9]|[12][0-9]|3[01])$/,
```

Comment:
This solution simply works. And it is taken from RegEx related resource. I think they do know what to do :-)
BTW, it works like a charm.
Link to that very resource: http://www.regular-expressions.info/dates.html